libraries/botframework-connector/src/auth/authValidator.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Activity, Channels, RoleTypes } from 'botframework-schema';
+import { AuthenticationConfiguration } from './authenticationConfiguration';
+import { AuthenticationConstants } from './authenticationConstants';
+import { ClaimsIdentity } from './claimsIdentity';
+import { ICredentialProvider } from './credentialProvider';
+import { JwtTokenExtractor } from './jwtTokenExtractor';
+import { UnauthorizedAccessError } from './unauthorizedAccessError';
+import { VerifyOptions } from 'jsonwebtoken';
+
+// AuthValidator is a function signature for logic that validates an HTTP auth header
+export type AuthValidator = (
+    credentials: ICredentialProvider,
+    authConfig: AuthenticationConfiguration,
+    authHeader: string,
+    activity: Partial<Activity>
+) => Promise<ClaimsIdentity>;
+
+// IdentityValidator is a function signature for logic that validates a set of JWT identity claims
+export type IdentityValidator = (
+    credentials: ICredentialProvider,
+    claimsIdentity: ClaimsIdentity,
+    activity: Partial<Activity>
+) => Promise<void>;
+
+/**
+ * Make an auth validator
+ *
+ * @param {VerifyOptions} verifyOptions options for verifying JWTs
+ * @param {string} openIdMetadata url to fetch open ID metadata
+ * @param {IdentityValidator} identityValidator the identity validator logic
+ * @returns {AuthValidator} an auth validator function
+ */
+export function makeAuthValidator(
+    verifyOptions: VerifyOptions,
+    openIdMetadata: string,
+    identityValidator: IdentityValidator
+): AuthValidator {
+    return async (credentials, authConfig, authHeader, activity) => {
+        if (!authHeader.trim()) {
```

Comment:
Here is all the logic that (I believe) is common to every type of auth validation, at least as they currently exist. JWT extraction parameters and other validation, along with claims validation, is then delegated to the consumer.